Shared/Samples/Create load report/CreateLoadReportView.Model.swift

```diff
@@ -0,0 +1,292 @@
+                let addLoadAttributeFunction = UtilityTraceFunction(
+                    functionType: .add,
+                    networkAttribute: loadAttribute,
+                    condition: serviceCategoryComparison
+                )
+                // Create function input and output condition.
+                traceConfiguration.addFunction(addLoadAttributeFunction)
+                traceConfiguration.outputCondition = serviceCategoryComparison
+                // Set to false to ensure that service points with incorrect phasing
+                // (which therefore act as barriers) are not counted with results.
+                traceConfiguration.includesBarriers = false
+                // Assign the trace configuration to trace parameters.
+                traceParameters.traceConfiguration = traceConfiguration
+                self.traceParameters = traceParameters
+            }
+        }
+        
+        /// When the utility network is loaded, create a `UtilityElement`
+        /// from the asset type to use as the starting location for the trace.
+        private func makeStartingLocation() -> UtilityElement? {
+            // Constants for creating the default starting location.
+            let networkSourceName = "Electric Distribution Device"
+            let assetGroupName = "Circuit Breaker"
+            let assetTypeName = "Three Phase"
+            let terminalName = "Load"
+            let globalID = UUID(uuidString: "1CAF7740-0BF4-4113-8DB2-654E18800028")!
+            
+            // Create a default starting location.
+            if let networkSource = utilityNetwork.definition?.networkSource(named: networkSourceName),
+               let assetType = networkSource.assetGroup(named: assetGroupName)?.assetType(named: assetTypeName),
+               let startingLocation = utilityNetwork.makeElement(assetType: assetType, globalID: globalID) {
+                // Set the terminal for the location. (For our case, use the "Load" terminal.)
+                startingLocation.terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName })
+                return startingLocation
+            } else {
+                return nil
+            }
+        }
+        
+        /// Gets the utility tier's trace configuration.
+        private func getTraceConfiguration() -> UtilityTraceConfiguration? {
+            // Get a default trace configuration from a tier in the network.
+            utilityNetwork
+                .definition?
+                .domainNetwork(named: "ElectricDistribution")?
+                .tier(named: "Medium Voltage Radial")?
+                .defaultTraceConfiguration
+        }
+        
+        func run() async throws {
+            statusText = "Creating load report…"
+            defer { statusText = nil }
+            
+            guard let phasesNetworkAttribute,
+                  let initialExpression,
+                  let traceParameters else { return }
```

Comment:
This is an early return and it didn't actually run anything or fail.  How does the caller of this function know about the failure?  Same question was posed above. I think this is more of a question for the samples in general. What do you want to happen in the case where these variables are not there?  You could make them preconditions, but then you would need to make sure the "run" button is not enabled if any of these three variables are nil.

Reply:
These early returns now throw an error to be presented to the user.

Reply:
This comment still applies. I will write up a small branch to demonstrate what my line of thinking is here.